gm4_note_block_interface/data/nbi/functions/click/sneak_loop.mcfunction

```diff
@@ -0,0 +1,9 @@
+#@s = player who played a note block while sneaking
+#runs from nbi:click/setup
+
+#removes 1 from raycast
+scoreboard players remove @s gm4_nbi_ray 1
+#tests for note_block
+execute if block ~ ~ ~ note_block align xyz positioned ~0.5 ~0.6 ~0.5 run function nbi:interface/change
+#runs the loop again
+execute unless block ~ ~ ~ note_block if score @s gm4_nbi_ray matches 1.. positioned ^ ^ ^0.01 run function nbi:click/sneak_loop
```

Comment:
Your raycast increments could probably be increased to `^0.1` without much impact to the behavior of the system, but reduce the number of recursive loops by an order of magnitude.

Reply:
I will have a look at that later on today 👍